### ISSUE TITLE
Unexpected character: '[' at line 17 col 91

### ISSUE THREAD

**Issue by hrsma2i** [python-poetry/poetry#1814]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **macOS Mojave 10.14.6**: 
- **1.0.0**: 
- pyproject.toml file:

```
[tool.poetry]
name = "wear-zozo"
version = "0.1.0"
description = ""
authors = ["hrsma2i <hrs.ma2i@gmail.com>"]

[tool.poetry.dependencies]
python = "^3.6"
numpy = "^1.18"
tqdm = "^4.41"
aiohttp = "^3.6"
asyncio = "^3.4"
Pillow = "^6.2"
jupyter = "^1.0"
mypy = "^0.761.0"
black = {version = "^18.3-alpha.0", allows-prereleases = true}
aiodl = {git = "git@github.com:hrsma2i/aiodl.git@04d916dcf2ac552827e3e8781e2d81301f0b9a1c"}[tool.poetry.dev-dependencies]
flake8 = "^3.7"

[build-system]
requires = ["poetry>=0.12"]
build-backend = "poetry.masonry.api"
```

## Issue

`poetry shell` doesn't work throwing the following error.

```
Unexpected character: '[' at line 17 col 91
```

**Comment by hrsma2i** [comment]
I solved this issue.

**Comment by mdengler** [comment]
@hrsma2i How did you solve this issue?  I am having a similar issue with `python -m poetry add flask` (and other packages besides `flask`).

**Comment by mdengler** [comment]
> @hrsma2i How did you solve this issue? I am having a similar issue with `python -m poetry add flask` (and other packages besides `flask`).

It appears if `pyproject.toml` does not end with a newline, this behavior happens.

**Comment by dyerdave-cvs** [comment]
> I solved this issue.

It would be great if you shared HOW you solved this issue.  I am still getting the same error on a square bracket and the newline character didn't resolve it.

**Comment by mlgarchery** [comment]
I encountered this issue twice. Each time I modified the pyproject.toml by hand using VSCode. The error points exactly to the line and column in the file where the space character is causing an issue.
I solved it by removing the spaces in the lines I was adding and adding them again.

I'm not sure to fully understand this (I'm thinking as I write) but I first generated the pyproject.toml using `poetry init` which encode the file in a one format. Editing it with VSCode, I used another one (mine is UTF-8, I may wrongly suppose `poetry init` used the same).

Correct me if I am wrong.

This resource may help https://stackoverflow.com/questions/10993612/how-to-remove-xa0-from-string-in-python
